pkg/virtctl/expand/expand.go

```diff
@@ -0,0 +1,152 @@
+package expand
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	yamlv3 "gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/tools/clientcmd"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+var (
+	cmd *cobra.Command
+
+	filePathProvided  bool
+	namespaceProvided string
+	outputProvided    string
+)
+
+const (
+	COMMAND_EXPAND = "expand"
+	YAML           = "yaml"
+	JSON           = "json"
+
+	fileFlag, fileFlagShort           = "file", "f"
+	namespaceFlag, namespaceFlagShort = "namespace", "n"
+	outputFlag, outputFlagShort       = "output", "o"
+)
+
+func ExpandCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd = &cobra.Command{
+		Use:     "expand (VM)",
+		Short:   "Expands a virtual machine.",
+		Example: usage(COMMAND_EXPAND),
+		Args:    templates.ExactArgs("vm", 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			expand := Command{clientConfig: clientConfig}
+			return expand.Run(args)
+		},
+	}
+	cmd.Flags().BoolVarP(&filePathProvided, fileFlag, fileFlagShort, false, "If present, the Virtual Machine spec in provided file will be expanded")
+	cmd.Flags().StringVarP(&namespaceProvided, namespaceFlag, namespaceFlagShort, "", "If present, the namespace scope for this CLI request")
+	cmd.Flags().StringVarP(&outputProvided, outputFlag, outputFlagShort, YAML, "Specify a format that will be used to display output.")
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+func usage(cmd string) string {
+	usage := fmt.Sprintf("  # %s a virtual machine called 'myvm':\n", cmd)
+	usage += fmt.Sprintf("  {{ProgramName}} %s kubevirt myvm\n", cmd)
+	return usage
+}
+
+func createVMObject(filePath string) (*v1.VirtualMachine, error) {
+	readFile, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(readFile), 1024)
+	vm := &v1.VirtualMachine{}
+	err = decoder.Decode(vm)
+	if err != nil {
+		return nil, err
+	}
+
+	return vm, nil
+}
+
+func printExpandedVM(outputFormat string, expandedVm *v1.VirtualMachine) error {
+	var formatedOutput []byte
+	var err error
+
+	switch outputFormat {
+	case JSON:
+		formatedOutput, err = json.MarshalIndent(expandedVm, "", " ")
+	case YAML:
+		formatedOutput, err = yamlv3.Marshal(expandedVm)
+	default:
+		return fmt.Errorf("not supported output format defined: %v", outputFormat)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	cmd.Print(string(formatedOutput))
+	return nil
+}
+
+type Command struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+func expandVirtualMachine(virtClient kubecli.KubevirtClient, args []string) error {
+	var expandedVm *v1.VirtualMachine
+	var err error
+
+	if filePathProvided {
```

Comment:
I sense abusing positional args like this could cause trouble in the future.

Reply:
I will change it to use `StringVarP` instead.